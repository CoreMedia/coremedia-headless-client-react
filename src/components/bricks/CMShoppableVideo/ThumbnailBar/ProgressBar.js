// @flow
import React from 'react';
import PropTypes from 'prop-types';

import ProgressBox from './ProgressBox';
import Bar from './Bar';

type Props = {
  posX: number,
  width: number,
};

const ProgressBar = ({ posX, width }: Props) => (
  <ProgressBox>
    <Bar posX={posX} width={width} />
  </ProgressBox>
);

ProgressBar.propTypes = {
  posX: PropTypes.number,
  width: PropTypes.number,
};

ProgressBar.defaultProps = {
  posX: 0,
  width: 0,
};

export default ProgressBar;
