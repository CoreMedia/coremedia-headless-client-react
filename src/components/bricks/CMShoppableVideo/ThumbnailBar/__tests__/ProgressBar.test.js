// @flow
import React from 'react';

import ProgressBar from '../ProgressBar';

describe('ProgressBar Component', () => {
  it('should render correctly with default props', () => {
    const wrapper = shallow(<ProgressBar />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly with custom props', () => {
    const wrapper = shallow(<ProgressBar posX={10} width={20} />);
    expect(wrapper).toMatchSnapshot();
  });
});
