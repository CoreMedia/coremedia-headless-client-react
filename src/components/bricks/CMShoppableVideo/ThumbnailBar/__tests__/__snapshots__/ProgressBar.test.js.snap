// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component should render correctly with custom props 1`] = `
<ProgressBox>
  <Bar
    posX={10}
    theme={
      Object {
        "name": "default",
      }
    }
    width={20}
  />
</ProgressBox>
`;

exports[`ProgressBar Component should render correctly with default props 1`] = `
<ProgressBox>
  <Bar
    posX={0}
    theme={
      Object {
        "name": "default",
      }
    }
    width={0}
  />
</ProgressBox>
`;
